bench/algorithm/http-server/2.dart

```diff
@@ -6,30 +6,30 @@ import 'dart:math';
 import 'package:shelf/shelf.dart';
 import 'package:shelf/shelf_io.dart' as shelf_io;
 
-final HttpClient _client = new HttpClient();
+final HttpClient _client = HttpClient();
 
 Future main(List<String> arguments) async {
-  final n = arguments.length > 0 ? int.parse(arguments[0]) : 500;
+  final n = arguments.isNotEmpty ? int.parse(arguments[0]) : 500;
   final port = 20000 + Random().nextInt(30000);
   // print(port);
 
   final serverReceivePort = ReceivePort();
   late final SendPort serverSendPort;
 
-  serverReceivePort.first.then((message) {
-    serverSendPort = message;
+  await serverReceivePort.first.then((message) {
```

Comment:
Looks like it's causing deadlock here, below code should work
```dart
final task = serverReceivePort.first.then((message) {
    serverSendPort = message as SendPort;
    serverSendPort.send(port);
  });

  await Isolate.spawn(_startServer, serverReceivePort.sendPort);
  await task;
```

Reply:
Fixed! thanks!